File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/news/NewsSourcesDao.java
package de.tum.in.tumcampusapp.component.ui.news;


import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import de.tum.in.tumcampusapp.component.ui.news.model.NewsSources;

@Dao
public interface NewsSourcesDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(NewsSources news);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<NewsSources> news);

    @Query("SELECT * FROM news_sources WHERE id < 7 OR id > 13 OR id=:selectedNewspread")
    List<NewsSources> getNewsSources(String selectedNewspread);

    @Query("SELECT * FROM news_sources WHERE id=:id")
    NewsSources getNewsSource(int id);

    @Query("DELETE FROM news_sources")
    void flush();
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/news/NewsSourcesDao.kt
package de.tum.`in`.tumcampusapp.component.ui.news


import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import de.tum.`in`.tumcampusapp.component.ui.news.model.NewsSources

@Dao
interface NewsSourcesDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(news: NewsSources)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(news: List<NewsSources>)

    @Query("SELECT * FROM news_sources WHERE id < 7 OR id > 13 OR id=:selectedNewspread")
    fun getNewsSources(selectedNewspread: String): List<NewsSources>

    @Query("SELECT * FROM news_sources WHERE id=:id")
    fun getNewsSource(id: Int): NewsSources

    @Query("DELETE FROM news_sources")
    fun flush()

}
